### ISSUE TITLE
Support py3.12

### ISSUE THREAD

**Issue by chenrui333** [twisted/constantly#29]
```
  ==> python3.12 -m pip --python=/home/linuxbrew/.linuxbrew/Cellar/magic-wormhole/0.13.0/libexec/bin/python install --verbose --no-deps --no-binary=:all: --ignore-installed --no-compile /tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0
  Using pip 23.2.1 from /home/linuxbrew/.linuxbrew/lib/python3.12/site-packages/pip (python 3.12)
  Processing /tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0
    Preparing metadata (setup.py): started
    Running command python setup.py egg_info
    /tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0/versioneer.py:467: SyntaxWarning: invalid escape sequence '\s'
      LONG_VERSION_PY['git'] = '''
    Traceback (most recent call last):
      File "<string>", line 2, in <module>
      File "<pip-setuptools-caller>", line 34, in <module>
      File "/tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0/setup.py", line 41, in <module>
        version=versioneer.get_version(),
                ^^^^^^^^^^^^^^^^^^^^^^^^
      File "/tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0/versioneer.py", line 1405, in get_version
        return get_versions()["version"]
               ^^^^^^^^^^^^^^
      File "/tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0/versioneer.py", line 1339, in get_versions
        cfg = get_config_from_root(root)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/tmp/magic-wormhole--constantly-20231017-7702-1ep4os7/constantly-15.1.0/versioneer.py", line 399, in get_config_from_root
        parser = configparser.SafeConfigParser()
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
    error: subprocess-exited-with-error
```

relates to https://github.com/Homebrew/homebrew-core/pull/151475

**Comment by glyph** [comment]
I think the solution here is to drop versioneer and do what other Twisted projects are doing (i.e., incremental and nothing else)

**Comment by adiroiban** [comment]
Happy to remove versioneer ... and maybe just use simple text values for versions , and nothing else :)

**Comment by graingert** [comment]
Fixed by https://github.com/twisted/constantly/pull/35

**Comment by BKPepe** [comment]
Any chance to have released a new version with support for Python 3.12?

**Comment by glyph** [comment]
@BKPepe sorry for the delay, but I have written #47 to hopefully move us along into the present day
